apps/idos-sdk-e2e/tests/04-grants.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { MetaMask, metaMaskFixtures, testWithSynpress } from "@synthetixio/synpress";
+import basicSetup from "./wallet-setup/basic.setup";
+
+const test = testWithSynpress(metaMaskFixtures(basicSetup));
+const { expect } = test;
+
+test.beforeEach(async ({ context, page }) => {
+  test.setTimeout(120000);
+
+  await context.clearCookies();
+  await page.evaluate(() => window.localStorage.clear());
+});
+
+test("should create a grant successfully", async ({ context, page, metamaskPage, extensionId }) => {
+  const metamask = new MetaMask(context, metamaskPage, basicSetup.walletPassword, extensionId);
+  await page.getByRole("button", { name: "Connect a wallet" }).click();
+  await page.getByRole("button", { name: "Metamask" }).click();
+  await metamask.switchAccount("Account 1");
+  await metamask.connectToDapp(["Account 1"]);
+  await page.waitForTimeout(2000);
+  await metamask.confirmSignature();
+
+  const list = page.locator("#credentials-list");
+
+  // Get the credential ID that is going to be shared.
+  const credentialId = (await list.getByRole("listitem").first().getAttribute("id")) as string;
+
+  await page.goto("https://idos-example-dapp-plaground.vercel.app/");
+  await context.clearCookies();
+  await page.evaluate(() => window.localStorage.clear());
+
+  await page.getByRole("checkbox", { name: "Credentials" }).click();
+
+  await page.getByRole("button", { name: "EVM" }).click();
+  await metamask.approveSwitchNetwork();
+  await metamask.switchAccount("Account 1");
+  await metamask.connectToDapp(["Account 1"]);
+  await metamask.confirmSignature();
+
+  await page.waitForTimeout(2000);
+
+  await page.getByText(credentialId).click();
+
+  const popupPromise = page.waitForEvent("popup");
+  const idOSButton = page.frameLocator("#idos-enclave-iframe").locator("#unlock");
+  await idOSButton.click();
+  const idOSPopup = await popupPromise;
```

Comment:
Why are we juggling around the await for this promise? Can we get rid of it? If not, can you add a comment explaining why your hand was forced?